scripts/irods/test/test_prep_genquery_iterator.py

```diff
@@ -127,7 +134,194 @@ def setUp(self):
                                      "/" + os.path.split(self.test_dir_path)[-1] )
 
 
-    # remove the next line when msiGetMoreRows always returns an accurate value for continueInx
+    @unittest.skipIf(plugin_name == 'irods_rule_engine_plugin-irods_rule_language', 'only applicable for python REP')
+    def test_query_objects_case_insensitive(self):
+
+        if not self.full_test : return
+        os.mkdir(self.dir_for_coll)
+        filename = "MyFiLe.TsT"
+        for dummy in range(3):
+            fullpath = os.path.join(self.dir_for_coll,filename)
+            open(fullpath,"wb").write(b"bytes")
+            filename = filename.lower() if dummy == 0 else filename.upper()
+        lib.execute_command ('tar -cf {} {}'.format(self.bundled_coll, self.dir_for_coll))
+        self.admin.assert_icommand('icd')
+        self.admin.assert_icommand('iput -f {}'.format(self.bundled_coll))
+        self.admin.assert_icommand('ibun -x {} .'.format(self.bundled_coll))
+        test_coll = self.test_admin_coll_path
+
+        with generateRuleFile(names_list = self.to_unlink,
+                              prefix = "query_case_insensitive_") as f:
+            rule_file = f.name
+            rule_text = dedent('''\
+            from genquery import *
+            def main(args,callback,rei):
+                q1 = Query(callback,'DATA_ID',
+                     "COLL_NAME = '{{c}}' and DATA_NAME = '{{d}}'".format(c='{test_coll}',d='{filename}'),
+                     case_sensitive=True)
+                q2 = Query(callback,'DATA_ID',
+                     "COLL_NAME = '{{c}}' and DATA_NAME = '{{d}}'".format(c='{test_coll}'.upper(),d='{filename}'.lower()),
+                     case_sensitive=False)
+                L= [ len([i for i in q]) for q in (q1,q2) ]
+                callback.writeLine('stdout', repr(L).replace(chr(0x20),''))
+            OUTPUT ruleExecOut
+            ''')
+            print(rule_text.format(**locals()), file=f, end='')
+
+        output, err, rc = self.admin.run_icommand("irule -F " + rule_file)
+        self.assertTrue(rc == 0, "icommand status ret = {r} output = '{o}' err='{e}'".format(r=rc,o=output,e=err))
+        self.assertEqual(output.strip(), "[1,3]")
+
+    @unittest.skipIf(plugin_name == 'irods_rule_engine_plugin-irods_rule_language', 'only applicable for python REP')
+    def test_query_objects(self):
+        if not self.full_test : return
+        os.mkdir(self.dir_for_coll)
+        interesting = [0,1,2,100,254,255,256,257,258,510,511,512,513,514]
+        octals = ['%04o'%x for x in interesting]
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Unused variable 'octals'](https://app.codacy.com/manual/trel/irods/pullRequest?prid=5016322)

Reply:
the variable is used in Python formatting string later in file